cql3/statements/select_statement.hh

```diff
@@ -175,9 +176,25 @@ public:
 };
 
 class indexed_table_select_statement : public select_statement {
+public:
+    // Struct which stores data related to global index,
+    // computed on the prepare stage
+    struct global_index_prepare_data {
+        // Column definition of computed token column in global index view
+        const column_definition& _token_cdef;
+
+        // Will global posting list partition be sliced according to base query TOKEN restriction
+        bool _is_token_partition_slice;
```

Comment:
If this class's invariant is `_is_token_partition_slice == bool(_posting_list_clustering_restrictions)`, maybe we should get rid of _is_token_partition_slice.  If the two can diverge, please explain in the blurb what that means.

Reply:
The idea behind separating those two things is that in the future more things could be moved to `prepare`. In such a case, `_posting_list_clustering_restrictions` could be used by other "cases" (not only `_is_token_partition_slice`) and `_is_token_partition_slice` would be one "case" only (`TOKEN`).

Reply:
Either leave that separation for that future or explain the current contract.